homeassistant/components/kodi/media_player.py

```diff
@@ -386,14 +393,25 @@ async def _async_ws_connect(self):
         try:
             await self._connection.connect()
             self._on_ws_connected()
-        except jsonrpc_base.jsonrpc.TransportError:
+        except (jsonrpc_base.jsonrpc.TransportError, CannotConnectError):
             _LOGGER.info("Unable to connect to Kodi via websocket")
             _LOGGER.debug("Unable to connect to Kodi via websocket", exc_info=True)
+            await self._clear_connection(False)
+
+    async def _ping(self):
+        try:
+            await self._kodi.ping()
+        except (jsonrpc_base.jsonrpc.TransportError, CannotConnectError):
+            _LOGGER.info("Unable to ping  Kodi via websocket")
+            _LOGGER.debug("Unable to ping Kodi via websocket", exc_info=True)
```

Comment:
Why twice ?

Reply:
Remainder from previous implementation. I'll remove it.